zeno/agents/kba/tool_kba_timeseries.py

```diff
@@ -0,0 +1,192 @@
+import json
+from datetime import datetime
+from enum import Enum
+from pathlib import Path
+from typing import Annotated, Dict, List, Optional
+
+import geopandas as gpd
+import pandas as pd
+from langchain_anthropic import ChatAnthropic
+from langchain_core.messages import HumanMessage, SystemMessage
+from langchain_core.tools import tool
+from langgraph.prebuilt import InjectedState
+from pydantic import BaseModel, Field, constr, validator
+
+from zeno.agents.kba.prompts import (
+    KBA_TS_COLUMN_SELECTION_PROMPT,
+    KBA_TS_INSIGHTS_PROMPT,
+)
+
+data_dir = Path("data/kba")
+kba_ts_data = pd.read_parquet(data_dir / "kba_timeseries_data.parquet")
```

Comment:
Did you already sync the data into S3?

Reply:
umm nope, need to check my `aws` access again :-(